src/init/features/dataconnect/index.ts

```diff
@@ -175,17 +175,17 @@ async function promptForService(setup: Setup, info: RequiredInfo): Promise<Requi
 async function promptForCloudSQLInstance(setup: Setup, info: RequiredInfo): Promise<RequiredInfo> {
   if (setup.projectId) {
     const instances = await cloudsql.listInstances(setup.projectId);
-    let choices = instances.map((i) => {
-      return { name: i.name, value: i.name, location: i.region };
-    });
+      if (instances.length) {
```

Comment:
I still see a bug here. It would still prompt "Which CloudSQL instance would you like to use?" when there is no instance available in the target region.

We want to do `choices.length` after it's filtered by region.

```
let choices = instances.map((i) => {
    return { name: i.name, value: i.name, location: i.region };
});
// If we've already chosen a region (ie service already exists), only list instances from that region.
choices = choices.filter((c) => info.locationId === "" || info.locationId === c.location);
if (choices.length) {
      const freeTrialInstanceId = await checkForFreeTrialInstance(setup.projectId);
      if (!freeTrialInstanceId) {
        choices.push({ name: "Create a new instance", value: "", location: "" });
      }
      info.cloudSqlInstanceId = await promptOnce({
        message: `Which CloudSQL instance would you like to use?`,
        type: "list",
        choices,
      });
```